pkg/virt-launcher/virtwrap/converter/vcpu/vcpu.go

```diff
@@ -454,18 +454,38 @@ func AdjustDomainForTopologyAndCPUSet(domain *api.Domain, vmi *v12.VirtualMachin
 		}
 	}
 
-	var emulatorThread uint32
+	var emulatorThreads []uint32
+	var emulatorThreadsCPUSet string
 	if vmi.Spec.Domain.CPU.IsolateEmulatorThread {
-		emulatorThread, err = cpuPool.FitThread()
-		if err != nil {
-			e := fmt.Errorf("no CPU allocated for the emulation thread: %v", err)
-			log.Log.Reason(e).Error("failed to format emulation thread pin")
-			return e
+		CPUManagerPolicyBetaOption := v12.CPUManagerPolicyBetaOptions(vmi.Annotations[v12.CPUManagerPolicyBetaOptionsAnnotation])
+		if cpuManagerPolicyBetaOptionFullpCPUsOnlyEnabled(CPUManagerPolicyBetaOption) {
+			availableCoreCPUTune, err := cpuPool.FitCores()
+			if err != nil {
+				e := fmt.Errorf("no Core allocated for the emulation thread: %v", err)
+				log.Log.Reason(e).Error("failed to format emulation thread pin")
+				return e
+			}
+			emulatorThreads, err = convertVCPUPinToCPUList(availableCoreCPUTune.VCPUPin)
+			if err != nil {
+				e := fmt.Errorf("failed to convert vCPUPin to CPUList: %v", err)
+				log.Log.Reason(e).Error("failed to format emulation thread pin")
+				return e
+			}
+		} else {
+			availableThread, err := cpuPool.FitThread()
+			if err != nil {
+				e := fmt.Errorf("no CPU allocated for the emulation thread: %v", err)
+				log.Log.Reason(e).Error("failed to format emulation thread pin")
+				return e
+			}
+			emulatorThreads = append(emulatorThreads, availableThread)
 		}
-		appendDomainEmulatorThreadPin(domain, emulatorThread)
+
+		emulatorThreadsCPUSet = convertCPUListToCPUSet(emulatorThreads)
```

Comment:
perhaps this could just stay a slice? it's expected as a slice everywhere including on pkg/virt-handler/vm.go:2835

Reply:
as discussed offline, on virt-hanlder it is taken from the domain - where it is set as a CPUSet then converted to a slice. 
This change on virt-launcher only aims to remove one redundant convert to CPUSet.